### PR TITLE
History: update on renaming/moving a folder

### PR DESCRIPTION
**Pull request by hius07**
Closes https://github.com/koreader/koreader/issues/10995.

(Doesn't update sdr with "dir" setting, but that's another story)

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/10999)

### CODE REVIEW

File: frontend/readhistory.lua [hunk 1]
```diff
@@ -199,6 +200,21 @@ function ReadHistory:updateItemByPath(old_path, new_path)
     end
 end
 
+--- Updates the history list after renaming/moving a folder.
+function ReadHistory:updateItemsByPath(old_path, new_path)
+    local history_updated
+    for i, v in ipairs(self.hist) do
+        local file, count = v.file:gsub(old_path, new_path)
+        if count == 1 then
+            self.hist[i].file = file
```
Comment (poire-z) [inline frontend/readhistory.lua:209]: About the :gsub(), I guess you should try to match at start, and limit the nb of replacements to 1 (4th parameter, or no need if matching at start).
So sick cases like renaming bits of `/mnt/onboard/Library/Fiction/mnt/onboard/Library/myfile.epub` are supported :)

**Review by poire-z — APPROVED** [review]

File: frontend/readhistory.lua [hunk 2]
```diff
@@ -199,6 +200,22 @@ function ReadHistory:updateItemByPath(old_path, new_path)
     end
 end
 
+--- Updates the history list after renaming/moving a folder.
+function ReadHistory:updateItemsByPath(old_path, new_path)
+    old_path = "^"..old_path
```
Comment (Frenzie) [inline frontend/readhistory.lua:205]: Oof, this was so confusing to me. I think it'd be much clearer straight in the gsub, heh. :-)

**Review by Frenzie — APPROVED** [review]

File: frontend/readhistory.lua (hunk 2 above)
Comment (hius07) [reply frontend/readhistory.lua:205]: > straight in the gsub

It's in the loop.

Comment (Frenzie) [reply frontend/readhistory.lua:205]: Or old_path_pattern instead of just old_path

**Pull request merged**